api/v1beta1/hostedcluster_types.go

```diff
@@ -1324,8 +1324,7 @@ type AWSRolesRef struct {
 	// 	  	"Effect": "Allow",
 	// 	  	"Action": [
 	// 	  		"kms:Decrypt",
-	// 	  		"kms:Encrypt",
-	// 	  		"kms:GenerateDataKey",
+	// 	  		"kms:ReEncrypt",
```

Comment:
This should be `kms:ReEncrypt*` there is no ReEncrypt action see the AWS [reference](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awskeymanagementservice.html) for KMS actions.

And we should only require `GenerateDataKeyPairWithoutPlaintext`

Reply:
> We should also update the [KMS provider policy](https://github.com/openshift/hypershift/blob/main/api/v1alpha1/hostedcluster_types.go#L1784) to have `GenerateDataKeyPairWithoutPlaintext`

This role is used for etcd encryption, I think it needs a different set of perms cc @enxebre

Reply:
I am following the docs [here ](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSEncryption.html#ebs-encryption-permissions) and it just mentions `kms:ReEncrypt`. Our e2e are also passing, so it does actually exist.
AWS docs seems to have some inconsistencies.